openmc/deplete/transfer_rates.py

```diff
@@ -181,8 +191,26 @@ def set_transfer_rate(self, material, elements, transfer_rate, transfer_rate_uni
         else:
             raise ValueError("Invalid transfer rate unit '{}'".format(transfer_rate_units))
 
-        for element in elements:
-            check_value('element', element, ELEMENT_SYMBOL.values())
-            self.transfer_rates[material_id][element] = transfer_rate / unit_conv, destination_material_id
+        for component in components:
+            current_components = self.transfer_rates[material_id].keys()
+            if bool(re.match(GNDS_NAME_RE, component)):
+                nuc_element = re.split(r'\d+', component)[0]
+                if nuc_element in current_components:
+                    raise ValueError('Cannot add transfer rate for nuclide '
+                                     f'{component} to material {material_id} '
+                                     f'where element {nuc_element} already has '
+                                     'a transfer rate.')
+            else:
+                check_value('component', component, ELEMENT_SYMBOL.values())
+                r = re.compile(component + '[0-9]{1,3}(_m[0-9])?')
+                element_nucs = list(filter(r.match, current_components))
```

Comment:
I personally think list comprehensions are easier to interpret than `list(filter(...))`. Also, it's sufficient to just search for the element name plus a single digit (`re.match` searches for a match at the start of the string)
```suggestion
                element_nucs = [c for c in current_components
                                if re.match(component + r'\d', c)]
```

Reply:
This is much cleaner!